src/server/transaction.cc

```diff
@@ -1432,6 +1432,11 @@ OpResult<KeyIndex> DetermineKeys(const CommandId* cid, CmdArgList args) {
     if (!absl::SimpleAtoi(num, &num_custom_keys) || num_custom_keys < 0)
       return OpStatus::INVALID_INT;
 
+    // Only ZDiff numkeys > 0
```

Comment:
So, I had a reason why I put this here. Basically moving the check later would cause a crash on `transaction.cpp:258` which I originally thought it is not an issue (because the crash was caused by hashing the `numkeys` argument). However, since you mentioned this, I put this on a debugger and it made me realize that there is a more general issue. I verified this by running the command `ZUNION 0 myset` which caused the exact same crash above. I expect that all the commands with layout `command numkeys`  trigger this bug (because there is eval etc which I haven't checked). All in all, I think this should be addressed (I put a comment on the code) on a separate PR because it would require: 

1. Check all the commands involved with this layout
2. Decide which part of the execution path should change (where is more appropriate to throw this error)

and these are beyond the scope of this PR

Reply:
nice catch ;)